### PR TITLE
Make debug configurable

### PR DESCRIPTION
**Pull request by jodh-intel**
Add config options to toggle debug for the shim and the runtime (we already have an option for the hypervisor and the proxy is handled separately).

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @iphutch - please can you review the doc changes on this PR?

**Comment by mcastelino** [comment]
@jodh-intel @jcvenegas with this PR we should be able to have both debug logs from proxy as well as the ability to console into the VM right? That closes a gap in our debug environment where we could have one or the other.

**Comment by jodh-intel** [comment]
Ah - the unit-tests are happy, but some of the tests in the `tests` repo aren't since the `--debug` flag has been zapped.

We could retain `--debug` as an override for whatever is specified in the config file: if `--debug` is set/cleared, enable/disable debug for all components that can have debug enabled via the config file. That would actually be a change to the current system though since the `--debug` flag currently only affects the runtime log level.

There is also the issue of the `DEFENABLEDEBUG` build variable (although that is currently broken - see https://github.com/clearcontainers/runtime/pull/590/files#r143512863).

@mcastelino - can you clarify the scope of `DEFENABLEDEBUG`? I'm not sure we want such a variable to apply to all debug options (although admittedly, the admin can disable them via the config file if they desire).

**Comment by jodh-intel** [comment]
Hi @mcastelino - I'm not 100% clear on what you're saying. This PR doesn't provide anything wrt proxy debug (as that is currently only settable via a cli option (although it would be great if it could also be toggled via an API call so the runtime could control debug logs on a per pod basis).

I haven't tested this branch with your debug console one.

File: README.md [hunk 1]
```diff
@@ -102,6 +117,14 @@ attempt to create it.
 It is the Administrator's responsibility to ensure there is sufficient
 space for the global log.
 
+### Enabling debug for various components
+
+The runtime, the shim (`cc-shim`), and the hypervisor all have separate debug
+options in the configuration file (see the [configuration](#Configuration) file).
```
Comment (iphutch) [inline README.md]: This reads as a little bit repetitive. Can we instead have the first "configuration" file as the link as it is on line 89?

Comment (jodh-intel) [reply README.md]: Good catch - fixed.

**Comment by grahamwhaley** [comment]
@jodh-intel I think if you were to have both the config file and command line ability to configure the debug, then you're going to need `--no-debug` or equivalent adding to the command line as well I think (and of course the matching tests).
It might be easier to just drop the command line and move to the config file. Only question I think would be if any developers would miss the simple ability to run items by hand with debug enabled on the command line, vs tweaking a line in the relevant config files. I'm fine with just config files I think...

**Comment by jodh-intel** [comment]
Thanks @grahamwhaley - agreed - dropping `--debug` entirely is my preference. I'll raise a PR to tweak the `tests` repo for this...

**Comment by jodh-intel** [comment]
Blocked on https://github.com/clearcontainers/tests/pull/601.

**Comment by jodh-intel** [comment]
@grahamwhaley - could you take a look please?

**Comment by sameo** [comment]
LGTM

**Pull request merged**